ui/app/utils/api-error-message.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { ErrorContext, ApiError } from 'vault/api';
+
+// accepts an error and returns error.errors joined with a comma, error.message or a fallback message
+export default async function (error: unknown, fallbackMessage = 'An error occurred, please try again') {
+  const messageOrFallback = (message?: string) => message || fallbackMessage;
+
+  if ((error as ErrorContext).response instanceof Response) {
+    const apiError: ApiError = await (error as ErrorContext).response?.json();
+
+    if (apiError.errors && typeof apiError.errors[0] === 'string') {
+      return apiError.errors.join(', ');
+    }
+    return messageOrFallback(apiError.message);
+  }
+
+  return messageOrFallback((error as Error)?.message);
+}
```

Comment:
This was derived from the existing `error-message` util. The functionality is the same but handles calling the `json` method on the response.

Reply:
Ha, I just left a comment above (now deleted) about handling. 
1. Could you rename this util to be more descriptive so it's easy by the name to see how it's different than the error-message util? 
2. Your comment here, can you also add that onto your comment on line 8? It's a helpful comment for when someone asks how is this different than the other util.

Reply:
So the plan is to remove the `error-message` util once Ember Data is gone. I should have communicated that. I started by updating `error-message` but then since the function needs to be async to accommodate the `json` method I went in this direction. Would capturing this in comments for both utils suffice or would you like me to rename? I would like to stay away from using open api in naming patterns though and stick with just api since that's what the service is what we are interacting with.

Reply:
capturing in comments is good enough for me. thanks for the comment.